src/app/housekeeper/services/housekeeper-amenidades.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import { environment } from '../../../environments/environment';
+import { Observable } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { map } from 'rxjs/operators';
+import { ChamberMaid, Maid } from '../interfaces/registerAmenidades.interface';
+import { Register } from '../interfaces/registerAmenidades.interface';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HousekeeperAmenidadesService {
+
+  private readonly URL: string = environment.URL;
+
+  constructor(private httpClient: HttpClient) { }
+
+  getMaid(): Observable<ChamberMaid[]> {
+    return this.httpClient.get<Maid>(`${this.URL}maid/list`).pipe(
+      map((response) => {
+        return response.body;
+      })
+    );
+  }
+
+  sendInfo(register: Register){
+    return this.httpClient.post(`${this.URL}housekeeper/register`, register).pipe(
```

Comment:
Tu pipe debes borrarlo si no lo ocupas o en todo caso puedes Inyectar ToastService y Router para que aquí uses un tap como operator y así lo borras de tu componente quedando así:
export class RegisterComponent {
onSubmit() {
  ......
  this.housekeeperAmenidades.sendInfo(register).subscribe();
}

Y aquí en tu servicio:

sendInfo(register: Register){
    return this.httpClient.post(`${this.URL}housekeeper/register`, register).pipe(
      tap(() => {
         this.router.navigate(['/housekeeper/registers']);
        this.toastService.success('¡Registro guardado exitosamente!');
     })
  )
}

De esta forma limpias más tu componente.